docs/ADR-72-ab-testing.md

```diff
@@ -0,0 +1,154 @@
+>
+> - assuming total: `1.000` users
+> - activation rollout (50%): `500` users
+> - `enabled` variant (50%): `250` users
+> - `disabled` variant (50%): `250` users
+>
+> For a total of `1.000` only half of then will count as testing users and half of those will see the `enabled` options, that mean that `750` users will see the old version but only `250` will be consider in the result of the A/B testing.
+> If your objective is to test `500` users with `enabled` and `500` users with `disabled` remove the rollout configuration.
+
+Once your A/B testing is ready you should get something like this:
+
+```json5
+{
+  "flags": {
+    "test-flag": true, // not an A/B testing
+    "test-test-feature_name_variant": true,
+  },
+  "variants": {
+    "test-feature_name_variant": {
+      "name": "enabled" // or "disabled"
+    }
+  }
+}
+```
+
+If instead your A/B testing is inactive you should get something like this:
+
+```json5
+{
+  "flags": {
+    "test-flag": true, // not an A/B testing
+  },
+  "variants": {}
+}
+```
+
+#### Including extra data
+
+Some times you need to include additional data to know how to handle each variant, this is useful because you can develop the variant implementation and detached from the end value (that can be content that is not ready jet) you will be able to develop and deploy you a/b testing and other team in charge of produce the content will be able to test it without any other interaction with your code.
+
+![Extra data](./resources/ADR-72/006-extra-data.png)
+
+### Segment: measuring your new A/B testing
+
+To measure if any of the variants of an A/B testing is significant we need to know three things:
+
+- the total number of users in the A/B testing
+- which variant got each one
+- the success metric
+
+To know the total number of users we send an event `feature_flags` each time a user gets a new set of feature flags, but to know which variant each user gets we need to format our feature flags as the following array:
+
+```js
+[
+  `${FEATURE_FLAG_NAME}`,
+  `${FEATURE_FLAG_NAME}:${FEATURE_FLAG_VARIANT_NAME}`,
+]
+```
+
+For example, the previous feature flags will be format as:
+
+```json5
+[
+  "test-flag", // not an A/B testing
+  "test-feature_name_variant",
+  "test-feature_name_variant:enabled",
+  // or "test-feature_name_variant:disabled",
+]
+```
+
+And send it in the prop `featureFlags` this way we can easy search feature flags and or variant.
+
+```ts
+analytics.track(
+  `feature_flags`,
+  {
+    featureFlags: [
+      "test-flag", // not an A/B testing
+      "test-feature_name_variant",
+      "test-feature_name_variant:enabled",
+    ]
+  }
+)
+```
+
+Finally you define your success metric (and time frame), this way the data team (or any other) will be abel to generate a dashboard to measure the success (or not) of your test.
+
+Additionally you can send all features in your others events, this way you can create more dashboards and filter only for events generated during your A/B testings
```

Comment:
I'd still remove this suggestion, i don't see the point in adding this and it adds a lot of data in every event

Reply:
oki